### PR TITLE
feat: pass list type parameters via client sdk

### PR DESCRIPTION
**Pull request by christinestraub**
The purpose of this PR is to support using the same type of parameters as `partition_*()` when using `partition_via_api()`. This PR works together with `unsturctured-api` [PR #368](https://github.com/Unstructured-IO/unstructured-api/pull/368).

**Note:** This PR will support extracting image blocks("Image", "Table") via partition_via_api().

### Summary
- update `partition_via_api()` to convert all list type parameters to JSON formatted strings before passing them to the unstructured client SDK
- add a unit test function to test extracting image blocks via `parition_via_api()`
- add a unit test function to test list type parameters passed to API via unstructured client sdk

### Testing
```
from unstructured.partition.api import partition_via_api

elements = partition_via_api(
    filename="example-docs/embedded-images-tables.pdf",
    api_key="YOUR-API-KEY",
    strategy="hi_res",
    extract_image_block_types=["image", "table"],
)

image_block_elements = [el for el in elements if el.category == "Image" or el.category == "Table"]
print("\n\n".join([el.metadata.image_mime_type for el in image_block_elements]))
print("\n\n".join([el.metadata.image_base64 for el in image_block_elements]))
```

### CODE REVIEW

**Review by cragwolfe — APPROVED** [review]

File: test_unstructured/partition/test_api.py [hunk 1]
```diff
@@ -210,6 +211,52 @@ def test_partition_via_api_valid_request_data_kwargs():
...
+        strategy="hi_res",
+        extract_image_block_types=["image", "table"],
+        api_key=get_api_key(),
+    )
+    image_elements = [el for el in elements if el.category == ElementType.IMAGE]
+    for el in image_elements:
+        assert el.metadata.image_base64 is not None
+        assert el.metadata.image_mime_type is not None
+        image_data = base64.b64decode(el.metadata.image_base64)
+        assert isinstance(image_data, bytes)
```
Comment (pawel-kmiecik) [inline test_unstructured/partition/test_api.py:227]: additionally, we could try to load the image with PIL and check some basic metadata to confirm it's a correct image file. But as this is a high-priority bug - we could do that in a separate PR.

**Review by pawel-kmiecik — APPROVED** [review]
LGTM - I've got one suggestion but it can be done in additional PR or dropped :)

File: test_unstructured/partition/test_api.py (hunk 1 above)
Comment (christinestraub) [reply test_unstructured/partition/test_api.py:227]: Sounds reasonable! I'll plan to do that in a separate PR.
